test/replication.test.js

```diff
@@ -291,6 +289,96 @@ describe('Replication / Change APIs', function() {
         }
       ], done);
     });
+
+    it('detects 3rd-party UPDATE made during replication', function(done) {
+      async.series([
+        function createModel(next) {
+          SourceModel.create({ id: '1' }, next);
+        },
+        function replicate(next) {
+          SourceModel.replicate(TargetModel, next);
+        },
+        function updateModel(next) {
+          SourceModel.updateAll({ id: '1' }, { name: 'source' }, next);
+        },
+        function replicateWith3rdPartyModifyingData(next) {
+          var bulkUpdate = TargetModel.bulkUpdate;
+          TargetModel.bulkUpdate = function(data, cb) {
+            var self = this;
+            // Simulate the situation when a 3rd party modifies the instance
+            // to be updated while a replication run is in progress,
+            // where the 3rd-party does not update change-tracking records.
+            TargetModel.dataSource.connector.updateAttributes(
+              TargetModel.modelName,
+              '1',
+              { name: '3rd-party' },
+              function(err) {
+                if (err) return cb(err);
+                bulkUpdate.call(self, data, cb);
+              });
+          };
+
+          SourceModel.replicate(
+            TargetModel,
+            function(err, conflicts, cps, updates) {
+              if (err) return next(err);
+
+              var conflictedIds = getPropValue(conflicts || [], 'modelId');
+              expect(conflictedIds).to.eql(['1']);
+
+              // resolve the conflict using ours
+              conflicts[0].resolve(next);
+            });
+        },
+
+        replicateExpectingSuccess(),
+        verifyInstanceWasReplicated(SourceModel, TargetModel, '1')
+      ], done);
+    });
+
+    it('detects 3rd-party CREATE made during replication', function(done) {
```

Comment:
I'll add the test as part of modifying the DELETE implementation as mentioned above.